qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_study_info(user, study_proc=None, proc_samples=None):
+    """builds list of dicts for studies table, with all html formatted
+
+    Parameters
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study.
+    proc_samples : dict of lists
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data.
+
+    Notes
+    -----
+    Both study_proc and proc_samples must be passed, or neither passed.
+    """
+    # Logic check to make sure both needed parts passed
+    build_info = False
+    if study_proc is not None and proc_samples is None:
+        raise IncompetentQiitaDeveloperError(
+            'Must pass proc_samples when study_proc given')
+    elif proc_samples is not None and study_proc is None:
+        raise IncompetentQiitaDeveloperError(
+            'Must pass study_proc when proc_samples given')
+    elif proc_samples is None:
+        build_info = True
+
     # get list of studies for table
     study_list = user.user_studies.union(
```

Comment:
I know that you did not change this, but it is probably worth changing: the name of this variable is misleading. It is called `study_list` but the variable is of type `set`. Renaming it to `study_set` will be  better to avoid future development burden. The main reason is that when I was reading the code I though "Lists do not have the union/intersection method", and then I realize that the variable is a set.